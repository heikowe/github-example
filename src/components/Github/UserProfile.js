import React from 'react';
import PropTypes from 'prop-types';

const UserProfile = ({bio}) => {
  return (
     <div className="col-md-4">
       <ul>
		   {bio.avatar_url && <li className="list-group-item"> <img src={bio.avatar_url} alt="avatar" className="img-rounded img-responsive"/></li>}
		   {bio.name && <li className="list-group-item">Name: {bio.name}</li>}
		   {bio.login && <li className="list-group-item">Username: {bio.login}</li>}
		   {bio.email && <li className="list-group-item">Email: {bio.email}</li>}
		   {bio.location && <li className="list-group-item">Location: {bio.location}</li>}
		   {bio.company && <li className="list-group-item">Company: {bio.company}</li>}
		   {bio.followers > 0 && <li className="list-group-item">Followers: {bio.followers}</li>}
		   {bio.following > 0 && <li className="list-group-item">Following: {bio.following}</li>}
		   {bio.public_repos && <li className="list-group-item">Public Repos: {bio.public_repos}</li>}
		   {bio.blog && <li className="list-group-item">Blog: <a href={bio.blog}> {bio.blog}</a></li>}
       </ul>
    </div>
  )
}

UserProfile.propTypes = {
  bio: PropTypes.object.isRequired
}

export default UserProfile
