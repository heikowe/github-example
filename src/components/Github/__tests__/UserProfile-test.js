import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import UserProfile from './../UserProfile';

const userProfileTestData = {
	"login": "maxmu",
	"avatar_url": "https://avatars0.githubusercontent.com/u/25481546?v=4",
	"name": "Max Mustermann",
	"email": "max@mustermann.de",
	"location": "Germany",
	"company": "Muster Company",
	"followers": 20,
	"following": 32
}

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<UserProfile bio={userProfileTestData}/>, div);
});

it('creates/compares snapshot for UserProfile', () => {
	const component = renderer.create(
		<UserProfile bio={userProfileTestData}/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});