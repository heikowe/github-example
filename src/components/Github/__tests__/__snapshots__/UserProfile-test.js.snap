// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates/compares snapshot for UserProfile 1`] = `
<div
  className="col-md-4"
>
  <ul>
    <li
      className="list-group-item"
    >
       
      <img
        alt="avatar"
        className="img-rounded img-responsive"
        src="https://avatars0.githubusercontent.com/u/25481546?v=4"
      />
    </li>
    <li
      className="list-group-item"
    >
      Name: 
      Max Mustermann
    </li>
    <li
      className="list-group-item"
    >
      Username: 
      maxmu
    </li>
    <li
      className="list-group-item"
    >
      Email: 
      max@mustermann.de
    </li>
    <li
      className="list-group-item"
    >
      Location: 
      Germany
    </li>
    <li
      className="list-group-item"
    >
      Company: 
      Muster Company
    </li>
    <li
      className="list-group-item"
    >
      Followers: 
      20
    </li>
    <li
      className="list-group-item"
    >
      Following: 
      32
    </li>
  </ul>
</div>
`;
